tests/rules/prefer-concat-latest-from.test.ts

```diff
@@ -1,120 +1,176 @@
 import { stripIndent } from 'common-tags'
+import { fromFixture } from 'eslint-etc'
 import path from 'path'
 import rule, {
   messageId,
-  messageIdSuggest,
 } from '../../src/rules/effects/prefer-concat-latest-from'
 import { ruleTester } from '../utils'
 
 ruleTester().run(path.parse(__filename).name, rule, {
   valid: [
-    `
-      export class Effect {
+    {
+      code: `
+      import { withLatestFrom } from 'rxjs'
+      export class EffectOk {
         effect$ = createEffect(
           () =>
             this.actions$.pipe(
               ofType(CollectionApiActions.addBookSuccess),
               concatLatestFrom(action => this.store.select(fromBooks.getCollectionBookIds)),
               switchMap(([action, bookCollection]) => {
-                return {}
+                return of({})
               })
             ),
         );
       }`,
+      options: [{ strict: true }],
+    },
+    `
+    import { withLatestFrom } from 'rxjs'
+    class EffectOk1 {
+      detail$ = createEffect(() => {
+        return this.actions.pipe(
+          ofType(ProductDetailPage.loaded),
+          concatMap((action) =>
+            of(action).pipe(withLatestFrom(this.store.select(selectProducts))),
+          ),
+          mergeMap(([action, products]) => {
+          })
+        )
+      })
+    }`,
     `
-      class Effect {
+    import { withLatestFrom } from 'rxjs'
+    class EffectOk2 {
+      detail$ = createEffect(() => {
+        return this.actions.pipe(
+          ofType(ProductDetailPage.loaded),
+          concatMap((action) =>
+            of(action).pipe(
+              withLatestFrom(this.store.select$(something), (one, other) => somethingElse())
+            ),
+          ),
+          mergeMap(([action, products]) => {
+            return of(products)
+          }),
+        )
+      })
+    }`,
+  ],
+  invalid: [
+    fromFixture(
+      stripIndent`
+      import { withLatestFrom } from 'rxjs'
+      class EffectNotOk1 {
+        effect$ = createEffect(() =>
```

Comment:
Something that I just noticed while messing with this tests is that currently we're looking for `ClassProperty` in this rule, however... what about this case?

```ts
@Injectable()
export class UserEffects {
  readonly load$: CreateEffectMetadata;

  constructor(private readonly actions$: Actions) {
    this.load$ = createEffect(() =>
      this.actions$.pipe(
        ofType(userActions.load),
        withLatestFrom((action) =>
          this.store.select(fromBooks.selectCollectionBookIds),
        ),
        switchMap(([action, bookCollection]) => of({})),
      ),
    );
  }
}
```

Note that this concern isn't related to the fixer and also not related to this rule only, but all the others that rely on `ClassProperty` (except, of course `no-effect-decorator` and `no-effect-decorator-and-creator`).

Reply:
Makes sense to me 👍